mailbox/tika/src/test/java/org/apache/james/mailbox/tika/CachingTextExtractorTest.java

```diff
@@ -0,0 +1,98 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.tika;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.function.Supplier;
+
+import org.apache.james.mailbox.extractor.ParsedContent;
+import org.apache.james.mailbox.extractor.TextExtractor;
+import org.junit.Before;
+import org.junit.Test;
+import org.testcontainers.shaded.com.google.common.collect.ImmutableMap;
+
+public class CachingTextExtractorTest {
```

Comment:
you don't test the weighter ? the eviction ?

Reply:
I'd need to reverse engineer Guava's cache to test the eviction, as it happens before the cache is full (expensive process, you know). 

I'd rather not do that.

Reply:
I don't think so, there are invariants you can easily test : 
Given a cache with a 100 MiB limit, if I insert 200 MiB of data, we should find at most half the inserted data.
Given a cache with a 100 MiB limit, if I insert 200 MiB of data and access the first inserted element, this element should never be recomputed.
Time-based eviction is IMO less sensitive if the size limit is respected.

Reply:
I failed to describe the test : I wanted to say, during the insertion of the 200 MiB of data, you keep accessing the first element frequently.